### PR TITLE
Feature/hocs 3258 stop logspam from export closed connections

### PR DESCRIPTION
**Pull request by gardnersam**
When the CSVPrinter throws an IOException this PR catches and rethrows so the process is halted, error logged and error response sent. There was also a lot of duplicated code in the class so this has been extracted where possible. The code to process the exception being thrown back up the chain was already in place but not used as exceptions were just caught and logged instead causing the logspam this PR fixes.

### CODE REVIEW

File: src/main/java/uk/gov/digital/ho/hocs/audit/export/ExportService.java [hunk 1]
```diff
@@ -209,31 +234,35 @@ public void auditSomuExport(LocalDate from, LocalDate to, OutputStream output, S
...
-                    SOMU_TYPE_EVENTS,
-                    caseTypeCode);
-
+        try (CSVPrinter printer = new CSVPrinter(outputWriter, CSVFormat.DEFAULT.withHeader(headers.toArray(new String[0])))) {
+            Stream<AuditData> data = getAuditDataStream(false, SOMU_TYPE_EVENTS, from, to, caseTypeCode);
             ExportDataConverter exportDataConverter = convert ? exportDataConverterFactory.getInstance() : null;
-
             data.forEach((audit) -> {
+                try {
+                    Thread.sleep(1000);
```
Comment (LiamMacP) [inline src/main/java/uk/gov/digital/ho/hocs/audit/export/ExportService.java:242]: Needed?

File: src/main/java/uk/gov/digital/ho/hocs/audit/export/ExportService.java [hunk 2]
```diff
@@ -49,6 +50,10 @@
     public static final String[] CORRESPONDENT_EVENTS = {"CORRESPONDENT_DELETED", "CORRESPONDENT_CREATED", "CORRESPONDENT_UPDATED"};
     public static final String[] EXTENSION_EVENTS = {"EXTENSION_APPLIED"};
     public static final String[] ALLOCATION_EVENTS = {"STAGE_ALLOCATED_TO_TEAM", "STAGE_CREATED", "STAGE_RECREATED", "STAGE_COMPLETED", "STAGE_ALLOCATED_TO_USER", "STAGE_UNALLOCATED_FROM_USER"};
+    private static class StreamBrokenException extends RuntimeException {
```
Comment (LiamMacP) [inline src/main/java/uk/gov/digital/ho/hocs/audit/export/ExportService.java:53]: I'd prefer for this to be in it's own class, but not a deal breaker.

File: src/main/java/uk/gov/digital/ho/hocs/audit/export/ExportService.java (hunk 1 above)
Comment (gardnersam) [reply src/main/java/uk/gov/digital/ho/hocs/audit/export/ExportService.java]: Nope, left over from testing, glad you caught that

File: src/main/java/uk/gov/digital/ho/hocs/audit/export/ExportService.java (hunk 2 above)
Comment (gardnersam) [reply src/main/java/uk/gov/digital/ho/hocs/audit/export/ExportService.java]: Moved to external file

**Review by LiamMacP — APPROVED** [review]
Looks good to me...

**Pull request merged**